web/src/main/java/pl/edu/agh/papaya/security/TokenRepository.java

```diff
@@ -0,0 +1,29 @@
+package pl.edu.agh.papaya.security;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+@Component
+class TokenRepository {
+
+    private Cache<Object, UserDetails> tokenCache =
+            CacheBuilder.newBuilder()
+                    .maximumSize(1000)
```

Comment:
Jeszcze tu o magic number checkstyle krzyczy

W ogóle o tę klasę zresztą ogólnie. Ja nie posiadam jakiegoś dogłębnego zrozumienia ` @Component`, więc nie wiem, czy wiem, dlaczego to jest/nie jest okej.

Jakby co: `@SuppressWarnings({"PMD.BeanMembersShouldSerialize"})`

Reply:
Możesz mi też wytłumaczyć, dlaczego to jest/nie jest okej, jeśli Ci się chce :stuck_out_tongue:

Reply:
No i jeszcze dodanie `transient` go ucisza, zgodnie z sugestią

Reply:
Wydaje mi się, że magiczne stałe tutaj nie pomagają za wiele (wręcz przeciwnie), no ale dodałem.

Cache zrobiłem też `transient` bo nie jestem pewny co do serializacji Cache z Guavy.